app/routers/login.py

```diff
@@ -0,0 +1,72 @@
+from typing import Optional, Union
+
+from app.dependencies import get_db, templates
+from app.internal.security.dependancies import (
+    User, current_user)
+from app.internal.security.ouath2 import (
+    authenticate_user, create_jwt_token, LoginUser)
+from fastapi import APIRouter, Depends, Request
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
+
+
+router = APIRouter(
+    prefix="",
+    tags=["/login"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/login")
+async def login_user_form(
+        request: Request, message: Optional[str] = "",
+        current_user: Union[User, None] = Depends(current_user)) -> templates:
+    """rendering login route get method"""
+    if current_user:
+        return RedirectResponse(url='/')
+    return templates.TemplateResponse("login.html", {
+        "request": request,
+        "message": message,
+        'current_user': current_user
+    })
+
+
+@router.post('/login')
+async def login(
+        request: Request,
+        next: Optional[str] = "/",
```

Comment:
We should check `next` is not malicious

Reply:
I don't really understand what do you mean. This is the requested url, all i' am trying to do is to 'remember' it, if it is under dependency, and if the user was not logged in when sending the request.
The goal is is simply to return the user directly to the requested route, after logging in.
In this case, next is a parameter  i created in ouath2.auth_exception_handler. I chose to use that name because it is consensus name for this parameter. should i change that?

Reply:
The `next` can be `http://attacker.com`, for example.

Please read about open redirection attack.